language/gpt3/dataset.py

```diff
@@ -0,0 +1,200 @@
+            ]
+        else:
+            prompts_tokens = [self.tokenizer.tokenize(prompt) for prompt in prompts]
+
+        # Now we have a list of list of tokens which each list has a different
+        # size. We want to extend this list to:
+        #   - incorporate the tokens that need to be generated
+        #   - make all the sequences equal length.
+        # Get the prompts length.
+        prompts_length = [len(prompt_tokens) for prompt_tokens in prompts_tokens]
+        # Get the max prompts length.
+        max_prompt_len = max(prompts_length)
+        # Number of tokens in the each sample of the batch.
+        samples_length = max_prompt_len + tokens_to_generate
+        # Now update the list of list to be of the same size: samples_length.
+        for prompt_tokens, prompt_length in zip(prompts_tokens, prompts_length):
+            padding_size = samples_length - prompt_length
+            prompt_tokens.extend([self.tokenizer.eod] * padding_size)
+
+        # Now we are in a structured format, we can convert to tensors.
+        # prompts_tokens_tensor = torch.LongTensor(prompts_tokens)
+        # prompts_length_tensor = torch.LongTensor(prompts_length)
+
+        return prompts_tokens, prompts_length
+
+    def encode_samples(self):
+        print("Encoding Samples")
+
+        total_samples = len(self.sources)
+
+        source_encoded_input_ids = []
+        source_encoded_input_id_lengths = []
+        # source_encoded_attn_masks = []
+
+        for i in range(total_samples):
+            if i % 100 == 0 and self.debug:
+                print("Sentence:")
+                print(self.sources[i])
+                print(
+                    "--------------------------------------------------------------------------------"
+                )
+            tokens, length = self.tokenize_prompts(
+                [self.sources[i]], self.gen_kwards.get("max_new_tokens", 128), None
+            )
+            # attn_mask = self._build_attention_mask(tokens)
+            source_encoded_input_ids.append(tokens)
+            # source_encoded_attn_masks.append(attn_mask)
+            source_encoded_input_id_lengths.append(length)
+            if i % 100 == 0 and self.debug:
+                print(f"Tokens: {tokens}")
+                print(f"Original length: {length}")
+                # input("...")
+
+        return (
+            source_encoded_input_ids,
+            # source_encoded_attn_masks,
+            source_encoded_input_id_lengths,
+        )
+
+    def LoadSamplesToRam(self, sample_list):
+        pass
+
+    def UnloadSamplesFromRam(self, sample_list):
+        pass
+
+    def __del__(self):
+        print("Finished destroying QSL.")
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--tokenizer-type", default="GPT2BPETokenizer")
+    parser.add_argument(
+        "--vocab-file",
+        default="/content/drive/MyDrive/MLCommons/notebook_data/GPT3/vocab.json",
```

Comment:
please share these files as well

Reply:
Just added a link to download these dowlhttps://github.com/mlcommons/inference/blob/db052d300c0b7b4674d60e1fd35eff150e2bccc3/language/gpt3/README.md#download-tokenizer-files